doc/architecture/decisions/0016-introduce-cas2-api-namespace.md

```diff
@@ -0,0 +1,44 @@
+# 16. Introduce CAS2 API namespace
+
+Date: 2023-09-13
+
+## Status
+
+Draft
+
+## Context
+
+At present there is a single set of API endpoints and controllers. For 
+example the `/applications` endpoint is handled by the  `ApplicationsController.applicationsGet()` 
+function which then branches 3 
+ways according to which of the 3 CAS services is calling the endpoint. (The 
+identity of the calling service is communicated by the consumer via the
+`X-Service-Name` header.)
+
+In "ADR 15 Handle Nomis users without inheritance in CAS2" we describe the 
+decision to extract CAS2-specific "User" and "Application" code out to 
+stand-alone classes outside of the inheritance hierarchy. We believe that 
+this will make the codebase clearer and more maintainable.
+
+Before this work begins we see an opportunity to bring further 
+benefits of clarity and explicitness by separating CAS2 endpoints into their 
+own namespace and implementing CAS2 controllers. For example, we'd have 2 
+applications endpoints, a new one for CAS2 and the existing one handling 
+CAS1 and CAS3 rather than all three services: 
+
+```
+/applications -> ApplicationsController (CAS1 and CAS3)
+/cas2/applications -> CAS2ApplicationsController
+```
+## Decision
+
+We will implement a CAS2-specific API namespace and controllers as outlined. 
+We anticipate that this will make development and maintenance faster by 
+clearly separating CAS2 functionality from that of the other services. 
+
+Before starting the work to support Nomis users we'll extract the existing 
+behaviour to CAS2 controllers and endpoints.
+
+## Consequences
+
+??? Can you see any downsides to this? What am I missing? 
```

Comment:
My biggest worry here is the risk of repetition. Namespacing in this way means we will have to reimplement a lot of controller boilerplate.

Reply:
I'm going to do a spike today to see what it involves. 

I'm hoping there won't be too much ceremony and config required. It's often the case that a bit of repetition can be less dangerous than a premature abstraction!